resources.management.tf

```diff
@@ -137,3 +136,34 @@ resource "azurerm_log_analytics_linked_service" "management" {
   ]
 
 }
+
+resource "azurerm_user_assigned_identity" "management" {
+  for_each = local.azurerm_user_assigned_identity_management
+
+  provider = azurerm.management
+  # Mandatory resource attributes
+  name                = each.value.template.name
+  location            = each.value.template.location
+  resource_group_name = each.value.template.resource_group_name
+
+  # Optional resource attributes
+  tags = each.value.template.tags
+
+  # Set explicit dependency on Resource Group deployment
+  depends_on = [
+    azurerm_resource_group.management,
```

Comment:
I think we still need the depends on here?

Reply:
> I think we still need the depends on here? 

I've made the assignment depend on the UAMI, during apply the UAMI will be created before the policies. During destroy the policies will be destroyed before the UAMI - have it got that right

[see here](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/blob/d25f0edee62fdcbc0046d3b3cf104fb05e676106/resources.policy_assignments.tf#L92-L93)

Reply:
Yes, sorry I got it the wrong way round!